Cron/EventsTopic.php

```diff
@@ -85,11 +88,30 @@ public function moveRowsToSync()
         }
 
         $idsMoved = [];
+        $idsFailed = [];
 
         // Capture all events that have been moved and add data to Sync table
+        // Additionally capture all events that have failed to move and continue processing remaining events
         foreach ($eventsData as $event){
             if ($event['event'] == 'Added To Cart') {
-                $event['payload'] = json_encode($this->replaceQuoteIdAndCategoryIds($event['payload']));
+                try {
+                    $event['payload'] = json_encode($this->replaceQuoteIdAndCategoryIds($event['payload']));
+                }
+                catch (\Exception $e) {
+                    // the payload was likely truncated, this will catch any indexing errors that occur during processing
+                    // defaults to a failed response and allows the other rows to continue syncing
+                    $this->_klaviyoLogger->log(sprintf("Unable to process Added to Cart data: %s", $e->getMessage()));
```

Comment:
Can we make this logged error unique from the saving to the events table so we can identify where we're going over the column limit?